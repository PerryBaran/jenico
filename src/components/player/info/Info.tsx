import { SongInfo } from '../../../Interface';
import style from './info.module.css';

interface Props {
    data: SongInfo[], 
    songIndex: number, 
    albumIndex: number
}

function Info(props: Props) {
    const { data, songIndex, albumIndex } = props;

    return (
        <div className={style.container}>
            <img src={data[albumIndex]?.art} alt={`${data[albumIndex]?.title} cover art`}/>
            <div>
                <h3>{data[albumIndex]?.title}</h3>
                <p>{data[albumIndex]?.songs[songIndex]?.name}</p>
            </div>
        </div>
    );
};

export default Info;