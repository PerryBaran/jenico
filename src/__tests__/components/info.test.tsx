import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Info from "../../components/player/info/Info";

describe("Info", () => {
  const validProps = {
    albumArt: "art",
    albumName: "album name",
    songName: "song name",
  };

  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(
      <Info
        albumArt={validProps.albumArt}
        albumName={validProps.albumName}
        songName={validProps.songName}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("corretly renders props", () => {
    render(
      <Info
        albumArt={validProps.albumArt}
        albumName={validProps.albumName}
        songName={validProps.songName}
      />
    );

    expect(
      screen.getByAltText(`${validProps.albumName} cover art`)
    ).toHaveAttribute("src", validProps.albumArt);
    expect(screen.getByText(validProps.albumName)).toBeInstanceOf(
      HTMLHeadingElement
    );
    expect(screen.getByText(validProps.songName)).toBeInTheDocument();
  });
});
