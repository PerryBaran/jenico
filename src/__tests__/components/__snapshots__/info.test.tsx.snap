// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="album name cover art"
      src="art"
    />
    <div>
      <h3>
        album name
      </h3>
      <p>
        song name
      </p>
    </div>
  </div>
</DocumentFragment>
`;
